### PR TITLE
refactor of resources module.

### PR DESCRIPTION
**Pull request by DhyanKPatel**

### CODE REVIEW

File: src/resources/resources.service.ts [hunk 1]
```diff
@@ -9,35 +8,59 @@ import { AddResources } from './dto/addResources.dto';
...
+  async addResources(dto: AddResources): Promise<StatusRO> {
+    try {
+      let { resourceName, addedBy, resourceNo, amount, paidBy } = dto;
+      const addResourcesData: ResourcesModel = await this.dbService.create(
+        this.resourcesModel,
+        { resourceName, userId: addedBy },
+      );
 
-    if (error) {
+      if (addResourcesData && Object.keys(addResourcesData).length > 0) {
```
Comment (32KAR) [inline src/resources/resources.service.ts:46]: You no need to check Object.keys() for create query. Remove condition.

File: src/configuration/configuration.service.ts [hunk 2]
```diff
@@ -1,34 +1,132 @@
...
+          transaction,
+        );
+
+      const findResourcesDetails: ResourcesDetailsModel =
+        await this.dbService.findOne(this.resourcesDetailsModel, {
+          resourceId,
+        });
+
+      if (
+        (configuration !== null && configuration !== undefined) ||
```
Comment (32KAR) [inline src/configuration/configuration.service.ts:106]: Remove all condition and only put condition like if((configuration) || (findResourcesDetails && findResourcesDetails.length > 0)

File: src/configuration/configuration.service.ts [hunk 3]
```diff
@@ -1,34 +1,132 @@
...
+        (findResourcesDetails !== null && findResourcesDetails !== undefined)
+      ) {
+        await this.dbService.update(
```
Comment (32KAR) [inline src/configuration/configuration.service.ts:109]: Store in variable and check condition.

File: src/configuration/configuration.service.ts [hunk 4]
```diff
@@ -39,34 +137,105 @@ export class ConfigurationService {
...
         },
-      )
-      .catch((err) => {
-        error = err;
-      });
+        { id: hrId },
+      );
+
+      if (configuration === 1) {
+        await this.dbService.update(
```
Comment (32KAR) [inline src/configuration/configuration.service.ts:213]: Check condition and give success message.

File: src/resources/resources.service.ts [hunk 5]
```diff
@@ -9,35 +8,59 @@ import { AddResources } from './dto/addResources.dto';
...
+        await this.dbService.create(this.resourcesDetailsModel, {
```
Comment (32KAR) [inline src/resources/resources.service.ts:47]: Check also this in condition.

File: src/resources/resources.service.ts [hunk 6]
```diff
@@ -48,137 +71,124 @@ export class ResourcesService {
...
+
+      const resourcesAllocationData: ResourcesAssignModel =
+        await this.dbService.create(this.resourcesAssignModel, {
+          ...dto,
+          assignBy: addedBy,
+        });
+
+      if (
+        resourcesAllocationData &&
+        Object.keys(resourcesAllocationData).length > 0
```
Comment (32KAR) [inline src/resources/resources.service.ts:106]: When you create no need to check this condition.

File: src/resources/resources.service.ts [hunk 7]
```diff
@@ -284,121 +311,79 @@ export class ResourcesService {
...
-          );
-        }
+        const addDamageData: ResourcesAssignModel = await this.dbService.create(
+          this.resourcesAssignModel,
+          {
+            ...damageData,
+          },
+        );
 
         if (addDamageData && Object.keys(addDamageData).length > 0) {
```
Comment (32KAR) [inline src/resources/resources.service.ts:353]: When you create no need to check this condition.

File: src/resources/resources.service.ts [hunk 8]
```diff
@@ -284,121 +311,79 @@ export class ResourcesService {
...
-            return HandleResponse(
-              HttpStatus.INTERNAL_SERVER_ERROR,
-              `${Messages.FAILED_TO} damaged resources.`,
-              undefined,
-              {
-                errorMessage: error.original.sqlMessage,
-                field: error.fields,
-              },
+          const [updateResourcesDetails]: number[] =
+            await this.dbService.update(
```
Comment (32KAR) [inline src/resources/resources.service.ts:355]: You need to check condition for this.

File: src/resources/resources.service.ts [hunk 9]
```diff
@@ -408,245 +393,188 @@ export class ResourcesService {
...
 
-    if (error) {
+      type ObjectType = {
+        totalDevicesData: number;
+        totalDevicesOfCurrentMonthData: number;
+        totalAllocatedDevicesData: number;
+        totalAllocatedDevicesOfCurrentMonthData: number;
+      };
+
+      const obj: ObjectType = {
```
Comment (32KAR) [inline src/resources/resources.service.ts:569]: You have to give specific name for variable no give like obj.

File: src/resources/resources.service.ts (hunk 8 above)
Comment (DhyanKPatel) [reply src/resources/resources.service.ts:355]: I have Already check this condition.Please check it.

File: src/resources/resources.service.ts [hunk 10]
```diff
@@ -9,35 +8,64 @@ import { AddResources } from './dto/addResources.dto';
...
+          this.resourcesDetailsModel,
+          {
+            userId: addedBy,
+            resourceNo,
+            amount,
+            paidBy,
+            resourceId: addResourcesData.id,
+          },
+        );
+        if (addResourceDetails || addResourcesData) {
```
Comment (32KAR) [inline src/resources/resources.service.ts:57]: No need to check condition for addResourcesData because of this check already you made in line no 46.

**Pull request merged**